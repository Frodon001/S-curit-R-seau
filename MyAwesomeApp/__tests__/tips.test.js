import React from 'react';
import renderer from 'react-test-renderer';
import Tips from '../components/View/tips/TipsView';
import {render, cleanup} from 'react-native-testing-library';

beforeAll(() => {
    jest.mock('@react-native-community/async-storage');
})

afterEach(cleanup);

it('renders correctly', async () => {
    const tree = renderer.create(<Tips />).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('<Tips />', () => {
    it('should match snapshot', () => {
        const rendered = render(<Tips text={'abacaba'} />).toJSON();

        expect(rendered).toMatchSnapshot();
    });

    it('should properly render the text', () => {
        const rendered = render(<Tips text={'abacaba'} />);
        const textComponent = rendered.getByTestId('text');

        expect(textComponent.props.children).toEqual('abacaba');
    });

    it('should render red text', () => {
        const rendered = render(<Tips text={'abacaba'} />);
        const textComponent = rendered.getByTestId('text');

        expect(textComponent.props.style).toMatchObject({color: 'red'});
    });

    it('should wrap text with a flexible wrapper', () => {
        const rendered = render(<Tips text={'abacaba'} />);
        const wrapperComponent = rendered.getByTestId('wrapper');

        expect(wrapperComponent.props.style).toMatchObject({flex: 1});
    });
});
