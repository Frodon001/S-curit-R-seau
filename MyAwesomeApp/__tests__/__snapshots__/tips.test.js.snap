// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tips /> should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="wrapper"
  >
    <Text
      style={
        Object {
          "color": "red",
        }
      }
      testID="text"
    >
      Vous pouvez aider les oiseaux de la nature et favoriser leur survie
            en les aidant à trouver facilement une alimentation adaptée et à
            garder un habitat approprié.
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "01",
          "saison": "Automne",
          "tips": "Dès l’arrivée des premiers froids, la nourriture commence à se raréfier pour les oiseaux.

Apportez de la nourriture régulièrement pour compenser la raréfaction des ressources alimentaires.",
        },
        Object {
          "id": "02",
          "saison": "Hiver",
          "tips": "Le gel et la neige limitent l’accès des oiseaux à l’eau et à la nourriture.

Proposez de la nourriture telle que les graines ou/et des boules de graisse pour aider les oiseaux à résister à la période hivernale.

Installez une baignoire à oiseaux et veillez à ce que l’eau ne soit pas gelée et toujours propre, changez l’eau régulièrement.",
        },
        Object {
          "id": "03",
          "saison": "Printemps",
          "tips": "Il y a 4 fois plus d’oiseaux qu’en hiver, donc les besoins en alimentation augmentent. C’est aussi le début de la période de nidification.

Remplissez régulièrement les mangeoires de nourriture.

Préparez des nichoirs pour offrir aux oiseaux un lieu adapté pour couver et élever leurs petits.",
        },
        Object {
          "id": "04",
          "saison": "Été",
          "tips": "La période de nidification se poursuit pour aboutir à la naissance des petits, qui vont avoir besoin d’une alimentation spécifique pour leur croissance.

De plus, avec la chaleur estivale, l’accès à l’eau pure, vitale pour l’oiseau, est de plus en plus difficile.

Apportez une alimentation complémentaire aux oiseaux qui n’ont pas assez de temps pour collecter la nourriture pour leurs petits et pour eux.

Installez une baignoire pour permettre aux oiseaux de se désaltérer et de se rafraîchir. Vous pouvez également mettre de l’eau dans une brique ou dans des couvercles de confitures…

",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "marginLeft": 5,
          "marginRight": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Automne
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Dès l’arrivée des premiers froids, la nourriture commence à se raréfier pour les oiseaux.

Apportez de la nourriture régulièrement pour compenser la raréfaction des ressources alimentaires.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Hiver
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Le gel et la neige limitent l’accès des oiseaux à l’eau et à la nourriture.

Proposez de la nourriture telle que les graines ou/et des boules de graisse pour aider les oiseaux à résister à la période hivernale.

Installez une baignoire à oiseaux et veillez à ce que l’eau ne soit pas gelée et toujours propre, changez l’eau régulièrement.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Printemps
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Il y a 4 fois plus d’oiseaux qu’en hiver, donc les besoins en alimentation augmentent. C’est aussi le début de la période de nidification.

Remplissez régulièrement les mangeoires de nourriture.

Préparez des nichoirs pour offrir aux oiseaux un lieu adapté pour couver et élever leurs petits.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Été
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            La période de nidification se poursuit pour aboutir à la naissance des petits, qui vont avoir besoin d’une alimentation spécifique pour leur croissance.

De plus, avec la chaleur estivale, l’accès à l’eau pure, vitale pour l’oiseau, est de plus en plus difficile.

Apportez une alimentation complémentaire aux oiseaux qui n’ont pas assez de temps pour collecter la nourriture pour leurs petits et pour eux.

Installez une baignoire pour permettre aux oiseaux de se désaltérer et de se rafraîchir. Vous pouvez également mettre de l’eau dans une brique ou dans des couvercles de confitures…


          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="wrapper"
  >
    <Text
      style={
        Object {
          "color": "red",
        }
      }
      testID="text"
    >
      Vous pouvez aider les oiseaux de la nature et favoriser leur survie
            en les aidant à trouver facilement une alimentation adaptée et à
            garder un habitat approprié.
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "01",
          "saison": "Automne",
          "tips": "Dès l’arrivée des premiers froids, la nourriture commence à se raréfier pour les oiseaux.

Apportez de la nourriture régulièrement pour compenser la raréfaction des ressources alimentaires.",
        },
        Object {
          "id": "02",
          "saison": "Hiver",
          "tips": "Le gel et la neige limitent l’accès des oiseaux à l’eau et à la nourriture.

Proposez de la nourriture telle que les graines ou/et des boules de graisse pour aider les oiseaux à résister à la période hivernale.

Installez une baignoire à oiseaux et veillez à ce que l’eau ne soit pas gelée et toujours propre, changez l’eau régulièrement.",
        },
        Object {
          "id": "03",
          "saison": "Printemps",
          "tips": "Il y a 4 fois plus d’oiseaux qu’en hiver, donc les besoins en alimentation augmentent. C’est aussi le début de la période de nidification.

Remplissez régulièrement les mangeoires de nourriture.

Préparez des nichoirs pour offrir aux oiseaux un lieu adapté pour couver et élever leurs petits.",
        },
        Object {
          "id": "04",
          "saison": "Été",
          "tips": "La période de nidification se poursuit pour aboutir à la naissance des petits, qui vont avoir besoin d’une alimentation spécifique pour leur croissance.

De plus, avec la chaleur estivale, l’accès à l’eau pure, vitale pour l’oiseau, est de plus en plus difficile.

Apportez une alimentation complémentaire aux oiseaux qui n’ont pas assez de temps pour collecter la nourriture pour leurs petits et pour eux.

Installez une baignoire pour permettre aux oiseaux de se désaltérer et de se rafraîchir. Vous pouvez également mettre de l’eau dans une brique ou dans des couvercles de confitures…

",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "marginLeft": 5,
          "marginRight": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Automne
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Dès l’arrivée des premiers froids, la nourriture commence à se raréfier pour les oiseaux.

Apportez de la nourriture régulièrement pour compenser la raréfaction des ressources alimentaires.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Hiver
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Le gel et la neige limitent l’accès des oiseaux à l’eau et à la nourriture.

Proposez de la nourriture telle que les graines ou/et des boules de graisse pour aider les oiseaux à résister à la période hivernale.

Installez une baignoire à oiseaux et veillez à ce que l’eau ne soit pas gelée et toujours propre, changez l’eau régulièrement.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Printemps
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            Il y a 4 fois plus d’oiseaux qu’en hiver, donc les besoins en alimentation augmentent. C’est aussi le début de la période de nidification.

Remplissez régulièrement les mangeoires de nourriture.

Préparez des nichoirs pour offrir aux oiseaux un lieu adapté pour couver et élever leurs petits.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <Text>
              Été
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <Text>
            La période de nidification se poursuit pour aboutir à la naissance des petits, qui vont avoir besoin d’une alimentation spécifique pour leur croissance.

De plus, avec la chaleur estivale, l’accès à l’eau pure, vitale pour l’oiseau, est de plus en plus difficile.

Apportez une alimentation complémentaire aux oiseaux qui n’ont pas assez de temps pour collecter la nourriture pour leurs petits et pour eux.

Installez une baignoire pour permettre aux oiseaux de se désaltérer et de se rafraîchir. Vous pouvez également mettre de l’eau dans une brique ou dans des couvercles de confitures…


          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
